tests/integration/targets/test_moving_objects_between_buckets/tasks/main.yml

```
@@ -0,0 +1,154 @@
+        bucket: "{{ bucket.src }}"
+        mode: list
+      register: bucket_src_info
+
+    - name: Check the source s3 bucket remaining objects
+      ansible.builtin.assert:
+        that:
+        - "bucket_src_info.s3_keys[0] == 'put-binary.bin'"
+        - "bucket_src_info.s3_keys[1] == 'test.py'"
+        
+    - name: Obtaining destination bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.dest }}"
+        mode: list
+      register: bucket_dest_info
+
+    - name: Check the destination s3 bucket obtained one object
+      ansible.builtin.assert:
+        that:
+        - "bucket_dest_info.s3_keys[0] == 'template/test.txt'"
+    
+    # Moving all objects between buckets =======================================================
+    - name: 
+      ansible.builtin.include_role:
+        name: cloud.aws_ops.moving_objects_between_buckets
+      vars:
+        source_bucket: "{{ bucket.src }}"
+        dest_bucket: "{{ bucket.dest }}"
+
+    - name: Obtaining source bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.src }}"
+        mode: list
+      register: bucket_src_info
+
+    - name: Check the source s3 bucket is empty
+      ansible.builtin.assert:
+        that:
+        - "bucket_src_info.s3_keys == []"
+        
+    - name: Obtaining destination bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.dest }}"
+        mode: list
+      register: bucket_dest_info
+
+    - name: Check the destination s3 bucket received all objects
+      ansible.builtin.assert:
+        that:
+        - "bucket_dest_info.s3_keys[0] in ['put-binary.bin','template/test.txt', 'test.py']"
+        - "bucket_dest_info.s3_keys[1] in ['put-binary.bin','template/test.txt', 'test.py']"
+        - "bucket_dest_info.s3_keys[2] in ['put-binary.bin','template/test.txt', 'test.py']"
+
+    # Delete source s3 bucket when empty ===================================================
+    - name: Delete source s3 bucket when empty
+      ansible.builtin.include_role:
+        name: cloud.aws_ops.moving_objects_between_buckets
+      vars:
+        source_bucket: "{{ bucket.src }}"
+        dest_bucket: "{{ bucket.dest }}"
+        delete_empty_source_bucket: True
+
+    - name: Obtaining source bucket
+      amazon.aws.s3_object:
+        bucket: "{{ bucket.src }}"
+        mode: list
+      register: bucket_src_info
+      ignore_errors: true
+    
+    - name: Check the destination s3 bucket received all objects
+      ansible.builtin.assert:
+        that: bucket_src_info is failed
+
+  always:
+    # Clean up starts here =================================================================
+    - name: Delete source s3 bucket
+      amazon.aws.s3_bucket:
+        name: "{{ bucket.src }}"
+        state: absent
+        force: true
+      ignore_errors: true
+
+    - name: Delete destination s3 bucket
+      amazon.aws.s3_bucket:
+        name: "{{ bucket.dest }}"
+        state: absent
+        force: true
+      ignore_errors: true
```

make sure to have a newline at end of each file, to avoid `ansible-lint` errors
```suggestion
      ignore_errors: true

```